pkg/proctree/proctree_feed.go

```diff
@@ -0,0 +1,246 @@
+				NsPPid:      -1, // do not change the ns parent PID
+				Uid:         -1, // do not change the parent UID
+				Gid:         -1, // do not change the parent GID
+			},
+			utils.NsSinceBootTimeToTime(feed.TimeStamp),
+		)
+		// try to enrich from procfs asynchronously (instead of reading procfs continuously)
+		pt.FeedFromProcFSAsync(int(feed.ParentPid))
+	}
+
+	parent.AddChild(feed.LeaderHash) // add the leader as a child of the parent
+
+	// Update the leader process (might already exist, might be the same as child)
+
+	leader, ok := pt.GetProcessByHash(feed.LeaderHash)
+	if !ok {
+		leader = pt.GetOrCreateProcessByHash(feed.LeaderHash)
+		leader.GetInfo().SetFeedAt(
+			TaskInfoFeed{
+				Name:        parent.GetInfo().GetName(),
+				Tid:         int(feed.LeaderTid),
+				Pid:         int(feed.LeaderPid),
+				NsTid:       int(feed.LeaderNsTid),
+				NsPid:       int(feed.LeaderNsPid),
+				StartTimeNS: feed.LeaderStartTime,
+				PPid:        int(feed.ParentPid),
+				NsPPid:      int(feed.ParentNsPid),
+				Uid:         0, // TODO: implement
+				Gid:         0, // TODO: implement
+			},
+			utils.NsSinceBootTimeToTime(feed.TimeStamp),
+		)
+		// try to enrich from procfs asynchronously (instead of reading procfs continuously)
+		pt.FeedFromProcFSAsync(int(feed.LeaderPid))
+	}
+
+	leader.SetParentHash(feed.ParentHash)
+
+	// Case 01: The child is a process (if leader == child, work is done)
+
+	if feed.ChildHash == feed.LeaderHash {
+		leader.GetExecutable().SetFeedAt(
+			parent.GetExecutable().GetFeed(),
+			utils.NsSinceBootTimeToTime(feed.TimeStamp),
+		)
+		leader.GetInterpreter().SetFeedAt(
+			parent.GetInterpreter().GetFeed(),
+			utils.NsSinceBootTimeToTime(feed.TimeStamp),
+		)
+		return nil
+	}
+
+	// Case 02: The child is a thread, and leader is the thread group leader.
+
+	thread := pt.GetOrCreateThreadByHash(feed.ChildHash)
+	thread.GetInfo().SetFeedAt(
+		TaskInfoFeed{
+			Name:        leader.GetInfo().GetName(),
+			Tid:         int(feed.ChildTid),
+			Pid:         int(feed.ChildPid),
+			NsTid:       int(feed.ChildNsTid),
+			NsPid:       int(feed.ChildNsPid),
+			StartTimeNS: feed.ChildStartTime,
+			PPid:        int(feed.ParentPid),
+			NsPPid:      int(feed.ParentNsPid),
+			Uid:         0, // TODO: implement
+			Gid:         0, // TODO: implement
+		},
+		utils.NsSinceBootTimeToTime(feed.TimeStamp),
+	)
+
+	thread.SetParentHash(feed.ParentHash) // all threads have the same parent as the thread group leader
+	thread.SetLeaderHash(feed.LeaderHash) // thread group leader is a "process" (not a thread)
+	leader.AddThread(feed.ChildHash)      // add the thread to the thread group leader
+
+	return nil
+}
+
+type ExecFeed struct {
+	TimeStamp         uint64
+	TaskHash          uint32
+	ParentHash        uint32
+	LeaderHash        uint32
+	CmdPath           string
+	PathName          string
+	Dev               uint32
+	Inode             uint64
+	Ctime             uint64
+	InodeMode         uint16
```

Comment:
Ditto - change "inter" into "interpreter" and "interpreter" into "interp" to match kernel names.

Reply:
![image](https://github.com/aquasecurity/tracee/assets/7395852/538e3491-939c-4ae2-92ed-cbd9baf653ae)